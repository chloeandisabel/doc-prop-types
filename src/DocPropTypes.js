const docProp = id => {
  const docProp = () => {
    return { id };
  };
  docProp.isRequired = () => {
    return { id: `${id}_IS_REQUIRED` };
  };
  return docProp;
};

const docPropWithArgs = id => {
  const docProp = args => {
    const reallyDocProp = () => {
      return { id, args };
    };
    reallyDocProp.isRequired = () => {
      return {
        id: `${id}_IS_REQUIRED`,
        args,
      };
    };
    return reallyDocProp;
  };

  return docProp;
};

export default {
  string: docProp("STRING"),
  number: docProp("NUMBER"),
  bool: docProp("BOOL"),
  symbol: docProp("SYMBOL"),
  func: docProp("FUNC"),
  object: docProp("OBJECT"),
  node: docProp("NODE"),
  element: docProp("ELEMENT"),
  array: docProp("ARRAY"),
  any: docProp("ANY"),
  shape: docPropWithArgs("SHAPE_OF"),
  oneOf: docPropWithArgs("ONE_OF"),
  oneOfType: docPropWithArgs("ONE_OF_TYPE"),
  arrayOf: docPropWithArgs("ARRAY_OF"),
  objectOf: docPropWithArgs("OBJECT_OF"),
  instanceOf: docPropWithArgs("INSTANCE_OF"),
  custom: docPropWithArgs("CUSTOM"),
};
