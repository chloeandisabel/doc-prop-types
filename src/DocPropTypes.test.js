import DocPropTypes from "./DocPropTypes";

describe("DocPropTypes", () => {
  it("mimics the official React.PropTypes API", () => {
    expect(DocPropTypes).toMatchSnapshot();
  });

  it("returns prop identifiers from its property functions", () => {
    const docPropsFired = Object.keys(DocPropTypes).reduce((fired, key) => {
      fired[key] = DocPropTypes[key]();
      return fired;
    }, {});

    expect(docPropsFired).toMatchSnapshot();
  });
});
