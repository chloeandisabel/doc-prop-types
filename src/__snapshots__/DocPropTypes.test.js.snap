// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocPropTypes mimics the official React.PropTypes API 1`] = `
Object {
  "any": [Function],
  "array": [Function],
  "arrayOf": [Function],
  "bool": [Function],
  "custom": [Function],
  "element": [Function],
  "func": [Function],
  "instanceOf": [Function],
  "node": [Function],
  "number": [Function],
  "object": [Function],
  "objectOf": [Function],
  "oneOf": [Function],
  "oneOfType": [Function],
  "shape": [Function],
  "string": [Function],
  "symbol": [Function],
}
`;

exports[`DocPropTypes returns prop identifiers from its property functions 1`] = `
Object {
  "any": Object {
    "id": "ANY",
  },
  "array": Object {
    "id": "ARRAY",
  },
  "arrayOf": [Function],
  "bool": Object {
    "id": "BOOL",
  },
  "custom": [Function],
  "element": Object {
    "id": "ELEMENT",
  },
  "func": Object {
    "id": "FUNC",
  },
  "instanceOf": [Function],
  "node": Object {
    "id": "NODE",
  },
  "number": Object {
    "id": "NUMBER",
  },
  "object": Object {
    "id": "OBJECT",
  },
  "objectOf": [Function],
  "oneOf": [Function],
  "oneOfType": [Function],
  "shape": [Function],
  "string": Object {
    "id": "STRING",
  },
  "symbol": Object {
    "id": "SYMBOL",
  },
}
`;
